akka-stream/src/main/scala/akka/stream/impl/CompletedPublishers.scala

```diff
@@ -40,6 +39,36 @@ private[akka] final case class ErrorPublisher(t: Throwable, name: String) extend
   override def toString: String = name
 }
 
+private[akka] final case class SingleElementPublisher[T](value: T, name: String) extends Publisher[T] {
+  import ReactiveStreamsCompliance._
+
+  private[this] class SingleElementSubscription(subscriber: Subscriber[_ >: T]) extends Subscription {
+    private[this] var done: Boolean = false
```

Comment:
Since it is only ever written to by the calling thread (the subscriber) the subscriber has to arrange for its own visibility between its own calls so I figured no such volatile was required.